autopush/metrics.py

```diff
@@ -122,16 +108,16 @@ def timing(self, name, duration, **kwargs):
 def from_config(conf):
     # type: (AutopushConfig) -> IMetrics
     """Create an IMetrics from the given config"""
-    if conf.datadog_api_key:
+    if conf.statsd_host and not isinstance(conf.statsd_host, mock.Mock):
```

Comment:
nit: I think my branch got away without this Mock check with this:

```
diff --git a/autopush/tests/test_db.py b/autopush/tests/test_db.py
index e814f69..2206d45 100644
--- a/autopush/tests/test_db.py
+++ b/autopush/tests/test_db.py
@@ -91,6 +91,7 @@ class DatabaseManagerTest(unittest.TestCase):
         fake_conf.message_table.tablename = "message_int_test"
         fake_conf.message_table.read_throughput = 5
         fake_conf.message_table.write_throughput = 5
+        fake_conf.statsd_port = 8125
         dm = DatabaseManager.from_config(
             fake_conf,
             resource=autopush.tests.boto_resource)
@@ -107,6 +108,7 @@ class DatabaseManagerTest(unittest.TestCase):
         fake_conf.message_table.tablename = "message_bogus"
         fake_conf.message_table.read_throughput = 5
         fake_conf.message_table.write_throughput = 5
+        fake_conf.statsd_port = 8125
         dm = DatabaseManager.from_config(
             fake_conf,
             resource=autopush.tests.boto_resource)
```

Reply:
Huh, perhaps. I do the mock check since I wasn't sure how many tests might not set that value.

Reply:
a better (?) solution might be to have a default `fake_conf` that we alter for local testing.